packages/jellyfish-api-core/__tests__/category/icxorderbook/submitDFCHTLC.test.ts

```diff
@@ -0,0 +1,328 @@
+    // maker fee should be reduced from accountDFIBeforeDFCHTLC
+    expect(Number(accountDFIAfterDFCHTLC[idDFI])).toStrictEqual(Number(accountDFIBeforeDFCHTLC[idDFI]) - Number(0.01))
+
+    // List htlc
+    const listHTLCOptions: ICXListHTLCOptions = {
+      offerTx: makeOfferTxId
+    }
+    const HTLCs: Record<string, ICXDFCHTLCInfo| ICXEXTHTLCInfo| ICXClaimDFCHTLCInfo> = await client.icxorderbook.listHTLCs(listHTLCOptions)
+    expect(Object.keys(HTLCs).length).toBe(2) // extra entry for the warning text returned by the RPC atm.
+    await checkDFCHTLCDetails(DFCHTLC, DFCHTLCTxId, HTLCs)
+  })
+
+  it('Should return an error when submiting DFCHTLC with invalid HTLC.offerTx', async () => {
+    const makeOfferTxId = await setupUntilDFIBuyOffer()
+    const accountDFIBeforeDFCHTLC = await container.call('getaccount', [accountDFI, {}, true])
+
+    // create DFCHTLC with invalid offerTx "123" - maker
+    const DFCHTLC: HTLC = {
+      offerTx: '123',
+      amount: new BigNumber(10), // in  DFC
+      hash: '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220',
+      timeout: 500
+    }
+    const promise = client.icxorderbook.submitDFCHTLC(DFCHTLC)
+
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('RpcApiError: \'offerTx (0000000000000000000000000000000000000000000000000000000000000123) does not exist\', code: -8, method: icx_submitdfchtlc')
+
+    const accountDFIAfterDFCHTLC = await container.call('getaccount', [accountDFI, {}, true])
+    // should be the same balance as accountDFIBeforeDFCHTLC
+    expect(Number(accountDFIAfterDFCHTLC[idDFI])).toStrictEqual(Number(accountDFIBeforeDFCHTLC[idDFI]))
+
+    // List htlc
+    const listHTLCOptions: ICXListHTLCOptions = {
+      offerTx: makeOfferTxId
+    }
+    const HTLCs: Record<string, ICXDFCHTLCInfo| ICXEXTHTLCInfo| ICXClaimDFCHTLCInfo> = await client.icxorderbook.listHTLCs(listHTLCOptions)
+    expect(Object.keys(HTLCs).length).toBe(1) // extra entry for the warning text returned by the RPC atm.
+  })
+
+  it('Should return an error when submiting DFCHTLC with incorrect HTLC.amount than amount in the offer', async () => {
+    const makeOfferTxId = await setupUntilDFIBuyOffer()
+    const accountDFIBeforeDFCHTLC = await container.call('getaccount', [accountDFI, {}, true])
+
+    // create DFCHTLC with invalid offerTx "123" - maker
+    const DFCHTLC: HTLC = {
+      offerTx: makeOfferTxId,
+      amount: new BigNumber(15), // in  DFC
+      hash: '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220',
+      timeout: 500
+    }
+    const promise = client.icxorderbook.submitDFCHTLC(DFCHTLC)
+
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test ICXSubmitDFCHTLCTx execution failed:\namount must be lower or equal the offer one\', code: -32600, method: icx_submitdfchtlc')
```

Comment:
Tests are too brittle? we should not compare the error string? if the C++ side changes only the error string for some reason, test maintenance time/effort would be higher.